fiftyone/__init__.py

```diff
@@ -25,9 +25,15 @@
 from fiftyone.__public__ import *
 
 import fiftyone.core.logging as _fol
-import fiftyone.migrations as _fom
 
-_fol.init_logging()
+# The old way of doing things, migrating database on import. If we
+#   REALLY need to do this, for example doc build, we can.
+if (
+    _os.environ.get("FIFTYONE_DISABLE_SERVICES", "0") != "1"
```

Comment:
@swheaton can you explain why the docs build is causing us problems here? Would love to avoid needing `_FIFTYONE_FORCE_DB_CONNECT_ON_IMPORT`

Reply:
Yeah I really hate it also 😫😫

Basically it is a mongoengine problem.

Autodoc tries to go in and touch every member and inherited member of each class. Which includes a field on the mongoengine document called "objects" which has a getter that requires the db connection 🤬

So it goes in, the connection has not been made, the "objects" getter fails, and then doc build fails.

I tried to play with autodoc settings including to ignore inherited members or specifically skip "objects", but didn't come up with anything that worked.

I'll try to think of other solutions, but brainstorming welcome

Reply:
I see. One way to clean up a bit would be to only call `migrate_database_if_necessary()` inside `establish_db_conn()`

Reply:
you tried added `exclude-members` [to the conf](https://github.com/voxel51/fiftyone/blob/85339fcecf03978435fed6fd94565d1c63f58dd6/docs/source/conf.py#L73
)? 


```py
autodoc_default_options = {
    ...
    "exclude-members": "objects"
}
```

Reply:
Ok perhaps I do not understand sphinx directives like I thought I did, but adding to `conf.py` works where directly to the automodule directive did not.
I thought I had tried that too but oh well glad it works!